components/plugins/webhook_notify.rb

```diff
@@ -0,0 +1,75 @@
+=begin
+    Copyright 2010-2016 Tasos Laskos <tasos.laskos@arachni-scanner.com>
+
+    This file is part of the Arachni Framework project and is subject to
+    redistribution and commercial restrictions. Please see the Arachni Framework
+    web site for more information on licensing and terms of use.
+=end
+
+# Send a webhook notification and payload after scan.
+#
+# @author Sean Handley <sean.handley@gmail.com>
+# @version 0.1
+class Arachni::Plugins::WebhookNotify < Arachni::Plugin::Base
+
+    def run
+        wait_while_framework_running
+    end
+
+    def clean_up
+        print_status 'Sending the notification...'
+
+        request = Arachni::HTTP::Request.new url:     options[:url],
+                                             method:  :post,
+                                             body:    interpolate_variables( options[:payload] ),
+                                             headers: { 'Content-Type' => options[:content_type] }
+        request.run
+
+        print_status 'Done.'
+    end
+
+    def interpolate_variables( data )
+        report       = framework.report
+        scanned_url  = framework.options.url
+        issues_found = report.issues.size
+        time_taken   = report.delta_time
+
+        data.gsub!( "$SCANNED_URL$", scanned_url )
+        data.gsub!( "$ISSUES_FOUND$", issues_found )
+        data.gsub!( "$TIME_TAKEN$", time_taken )
+    end
+
+    def self.info
+        {
+            name:        'Webhook notify',
+            description: %q{Sends a webhook payload over HTTP/HTTPS at the end of the scan.
+
+Valid payload variables to use in the payload:
+
+$SCANNED_URL$
+$ISSUES_FOUND$
+$TIME_TAKEN$
+                },
+            author:      'Sean Handley <sean.handley@gmail.com>',
+            version:     '0.1',
+            options:     [
+                Options::String.new( :url,
```

Comment:
`Options::URL` would be a better type, it'll verify the host and ensure a valid URL in general.